arcv: Fix missing vsetvli generation for arcv_vmv instructions.

luismgsilva opened the pull request:
The arcv_vmv RTL patterns were not generating vsetvli instructions because they lacked the proper RVV predication structure.  This caused the vector configuration to not be set up before executing the arcv.vmv.v.s and arcv.vmv.s.v instructions.

This patch restructures the arcv_vmv patterns to follow the standard RVV predicated instruction format by:
- Wrapping the instruction in an if_then_else with UNSPEC_VPREDICATE.
- Adding VL_REGNUM and VTYPE_REGNUM registers to the predication.
- Moving the actual instruction logic to an inner unspec.

This structure matches other ARCV vector instructions (like vnorm) and enables the vsetvl pass to properly insert vsetvli instructions before the arcv_vmv operations.

Commit:
arcv: Fix missing vsetvli generation for arcv_vmv instructions.

The arcv_vmv RTL patterns were not generating vsetvli instructions
because they lacked the proper RVV predication structure.  This caused
the vector configuration to not be set up before executing the
arcv.vmv.v.s and arcv.vmv.s.v instructions.

This patch restructures the arcv_vmv patterns to follow the standard
RVV predicated instruction format by:
- Wrapping the instruction in an if_then_else with UNSPEC_VPREDICATE.
- Adding VL_REGNUM and VTYPE_REGNUM registers to the predication.
- Moving the actual instruction logic to an inner unspec.

This structure matches other ARCV vector instructions (like vnorm)
and enables the vsetvl pass to properly insert vsetvli instructions
before the arcv_vmv operations.

Signed-off-by: Luis Silva <luiss@synopsys.com>

MichielDerhaeg reviewed (changes requested):
Add tests, check that the mode is correct

MichielDerhaeg commented:
> Add tests, check that the mode is correct

And the other tests?

luismgsilva commented:
> > Add tests, check that the mode is correct
> 
> And the other tests?

I've also updated the `arcv-vdsp-vmvi_v_s-compile-1.c`. Is there anything else I'm missing?

MichielDerhaeg commented:
> > > Add tests, check that the mode is correct
> > 
> > 
> > And the other tests?
> 
> I've also updated the `arcv-vdsp-vmvi_v_s-compile-1.c`. Is there anything else I'm missing?

There is also vmv(i)_s_v

luismgsilva commented:
> > > > Add tests, check that the mode is correct
> > > 
> > > 
> > > And the other tests?
> > 
> > 
> > I've also updated the `arcv-vdsp-vmvi_v_s-compile-1.c`. Is there anything else I'm missing?
> 
> There is also vmv(i)_s_v

Both tests have been updated. Also rebased on top of the new changes on `arc-2025.09` branch
